### PR TITLE
Fix incompatibility with Translatable

### PR DESCRIPTION
**Pull request by JorisDebonnet**
Before, children would always take the default Locale. This fixes that.

### CODE REVIEW

**Comment by micmania1** [comment]
I wonder if we could add an extension into CMSEditLink so that translatable can handle this itself. http://api.silverstripe.org/3.1/source-class-SiteTree.html#556

The problem with the current solution is that it only fixes the issue for Lumberjack and not any other module which might do similar. It also creates a dependency (although not really because of the if statement) which i'd like to avoid.

I should get time to look at this more on Friday, but its likely that this will get merged as I doubt an ideal solution is possible quite yet.

**Comment by tractorcow** [comment]
Or could you set the locale via updateDefaults() on sitetree?

At the moment it's set via "defaults" config fixed to the default locale. Change it to set the current locale in Translatable::populateDefaults() and boom fixed. :D No new extension points needed.

**Comment by tractorcow** [comment]
I concur with @micmania1 ; This is a translatable issue not a lumberjack one.

**Comment by JorisDebonnet** [comment]
Well I also agree; as long as it gets fixed somehow, I'll be happy ;)

**Comment by JorisDebonnet** [comment]
As this issue has been closed:
Has this in fact been fixed in Translatable then, or is there an issue about it there?

**Comment by micmania1** [comment]
Nope, this hasn't been fixed yet. I'm planning on looking at this tomorrow so i'l re-open until this is confirmed as fixed in translatable.

**Comment by dhensby** [comment]
Sorry - I closed it because this is mis-assigned. Can someone open an issue against the correct module, please?

**Comment by micmania1** [comment]
I've fixed this in translatable and cms https://github.com/silverstripe/silverstripe-translatable/issues/230

To avoid bumping minimum cms and translatable version I'd be happy to merge this with only a slight change. Can you change the if so that it matches the removed CMS check here (without isset on $data['Locale']): https://github.com/silverstripe/silverstripe-cms/pull/1411/files#diff-5a31043a41c8be646444a6d319aa6e37L151

Also, can you comment the code referring to this thread and give a little context as to why we need to refer to Translatable.

File: code/forms/gridfield/GridFieldSiteTreeAddNewButton.php [hunk 1]
```diff
@@ -119,6 +119,9 @@ public function handleAction(GridField $gridField, $actionName, $arguments, $dat
 				"ParentID" => $tmpData['currentPageID'],
 				"PageType" => $tmpData['pageType']
 			);
+			if(class_exists('Translatable')) {
```
Comment (micmania1) [inline code/forms/gridfield/GridFieldSiteTreeAddNewButton.php]:
```
if(class_exists('Translatable') && singleton('SiteTree')->hasExtension('Translatable')) { 
    // $data 
}
```

I think that should be okay

**Comment by JorisDebonnet** [comment]
This look good? 
If we know which version of Translatable makes this fix obsolete, I could add that in the comment.

**Comment by JorisDebonnet** [comment]
I figured it doesn't need the `class_exists('Translatable')` check if we're already checking whether SiteTree has that Extension available. I tried it: I can still create new blog posts via the Add button after I removed the translatable module from my test.

**Comment by micmania1** [comment]
Can you update the comment to link to this pull request. Also, the translatable module doesn't currently do it, but it will in the future so this comment won't actually be true anymore.

**Comment by JorisDebonnet** [comment]
Yeah, I know, that's why I wanted to comment something about when it will be available in Translatable. By updating the comment to link to this pull request, you mean in the actual php code? Or in the commit message?

**Comment by micmania1** [comment]
Either would do. Preferably both so its as clear as possible :+1:

**Comment by JorisDebonnet** [comment]
Added the comment, if it makes any sense... if it doesn't, tell me exactly what to write :p

**Comment by micmania1** [comment]
Thanks @JorisDebonnet

**Pull request merged**